### PR TITLE
Use build stages in Travis CI

### PR DESCRIPTION
**Pull request by davehunt**
Introduce build stages, including deploying to PyPI when the tests pass and a tag is pushed. This also fixes the ESR tests, which were incorrectly using release.

### CODE REVIEW

**Comment by davehunt** [comment]
This *should* fix #108 by doing a deploy only when all tests pass. I also reintroduced tests for ESR, which I mistakenly removed in an earlier commit and had to include a couple of fixes. We'll only know if a deploy works when we push a tag, but I'm not overly confident. Usually when a deploy condition is not met there is a message to indicate why a deployment doesn't happen. I don't see this, but maybe it's a quirk of build stages. I've asked about it in #travis on IRC.

File: foxpuppet/windows/browser/notifications/addons.py [hunk 1]
```diff
@@ -44,26 +44,27 @@ def addon_name(self):
...
+                self.root.find_anonymous_element_by_attribute(
+                    'anonid', 'button').click()
+            else:
+                self.root.find_element(*self._confirm_locator).click()
 
 
 class AddOnInstallComplete(BaseNotification):
     """Add-on install complete notification."""
 
-    def close(self):
```
Comment (jrbenny35) [inline foxpuppet/windows/browser/notifications/addons.py]: Can you explain why you removed this?

File: tox.ini [hunk 2]
```diff
@@ -1,5 +1,5 @@
 [tox]
-envlist = py{27,36}, flake8
+envlist = py{27,36}, flake8, docs
```
Comment (jrbenny35) [inline tox.ini]: How were the docs being built before?

Comment (davehunt) [inline tox.ini]: Good question, possibly tox ignores a missing environment in the list if there's a section for it.

**Comment by jrbenny35** [comment]
Looks good!

**Comment by jrbenny35** [comment]
> This should fix #108 by doing a deploy only when all tests pass. I also reintroduced tests for ESR, which I mistakenly removed in an earlier commit and had to include a couple of fixes. We'll only know if a deploy works when we push a tag, but I'm not overly confident. Usually when a deploy condition is not met there is a message to indicate why a deployment doesn't happen. I don't see this, but maybe it's a quirk of build stages. I've asked about it in #travis on IRC.

Was there any response on this?

**Comment by davehunt** [comment]
> Was there any response on this?

Unfortunately not. I'm going to go ahead and merge, if we have issues with the deploy we can raise an issue. The build stages feature is in beta, but I wouldn't want to do an automated deploy without it as it ensures all tests are passing rather than just one.

**Comment by davehunt** [comment]
"Skipping a deployment with the pypi provider because this is not a tagged commit" sounds good. Feel free to push a tag @jrbenny35 to see if we get a deployment! :)

**Pull request merged**